src/components/fields/checkbox-group/checkbox-group.tsx

```diff
@@ -107,7 +108,7 @@ export const CheckboxGroup = (props: IGenericFieldProps<ICheckboxGroupSchema>) =
 							data-testid={TestHelper.generateId(id, "checkbox")}
 							id={checkboxId}
 							disabled={disabled ?? option.disabled}
-							name={option.label}
+							name={sanitize(option.label)}
```

Comment:
name could take `option.value` instead since it represents the field value, rather than the display value

Reply:
decided to assign the unique id for the names instead, have applied the change to radio buttons as well